pettingzoo/utils/gymnasium_conversions.py

```diff
@@ -0,0 +1,138 @@
+"""
+Wrapping PettingZoo environments into the matching Gymnasium environments for specific agent by providing
+the action for all other agents.
+"""
+
+from typing import Callable, Optional
+import gymnasium
+from gymnasium.core import ActType, ObsType
+from pettingzoo import AECEnv, ParallelEnv
+
+
+# The first parameter is an agent (its identification), second parameter is the relevant observation.
+# The callable is expected to return the action that the given agent would like to take.
+ActOthers = Callable[[str, ObsType], ActType]
+
+
+class aec_to_gymnasium(gymnasium.Env):
+  """
+  Wraps a PettingZoo environment (AECEnv) and make a Gymnasium environment out of it (for a single agent).
+  In order to make it work, one needs to provide a mechanism to get the actions of all other agents.
+
+  Args:
+    aec_env: is the PezttingZoo AECEnv environment to wrap.
+    external_agent: is the agent for which the Gymnasium 'step' function is called.
+      In other words, this is the agent that would interact with this Gymnasium environment.
+    act_other: is a callable that accepts an agent and a
+      relevant observation and returns the action that should be taken on behalf of that agent.
+ 
+  Returns:
+    A Gymnasium environment.
+  """
+
+  def __init__(
+    self,
+    aec_env: AECEnv,
+    external_agent: str,
+    act_others: ActOthers
+  ):
+    super().__init__()
+    self._aec_env = aec_env
+    self._external_agent = external_agent
+    self._act_others = act_others
+    self.observation_space = self._aec_env.observation_space(external_agent)
+    self.action_space = self._aec_env.action_space(external_agent)
+
+  def reset(
+    self,
+    seed: Optional[int] = None,
+    options: Optional[dict] = None,
+  ):
+    super().reset(seed=seed)
+    self._aec_env.reset(seed=seed)
+    self._loop_others()
+    observation, _, _, _, info = self._aec_env.last()
+    return observation, info
+
+  def step(self, action):
+    agent = self._aec_env.agent_selection
+    assert agent == self._external_agent, f"expected it to be my turn, got {agent}"
```

Comment:
Can this ever happen? What should the user do if this appears?

Reply:
It should not happen as far as I can tell. But then maybe I'm wrong. This is why I would be happy to have an assert here.